### PR TITLE
AllFilter with OR, bug fix

### PR DESCRIPTION
**Pull request by Yuqi-Du**
**What this PR does**:
bug: when ALL operator is used inside a OR different condition of ALL are concatenated with OR. 
AllFilter should to be AND together seperately

**Which issue(s) this PR fixes**:
Fixes [#749](https://github.com/stargate/jsonapi/issues/749)

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/DBFilterBase.java [hunk 1]
```diff
@@ -610,8 +610,30 @@ JsonNode asJson(JsonNodeFactory nodeFactory) {
     boolean canAddField() {
       return false;
     }
+
+    public List<BuiltCondition> getAll() {
+      List<Object> values = arrayValue;
```
Comment (maheshrajamani) [inline src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/DBFilterBase.java:615]: Do we need this temp variable?

**Review by maheshrajamani — APPROVED** [review]

**Comment by maheshrajamani** [comment]
@Yuqi-Du Can you add a test to check the cql generated has brackets correctly for OR and ALL together?

**Comment by Yuqi-Du** [comment]
> @Yuqi-Du Can you add a test to check the cql generated has brackets correctly for OR and ALL together?

fixed.

File: src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/DBFilterBase.java (hunk 1 above)
Comment (Yuqi-Du) [reply src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/DBFilterBase.java]: solved

**Review by vkarpov15 — COMMENTED** [review]
LGTM :+1:

**Pull request merged**